pqarrow/arrowutils/merge.go

```diff
@@ -14,11 +15,15 @@ import (
 
 // MergeRecords merges the given records. The records must all have the same
 // schema. orderByCols is a slice of indexes into the columns that the records
-// and resulting records are ordered by. Note that the given records should
-// already be ordered by the given columns.
+// and resulting records are ordered by. While ordering the limit is checked before appending more rows.
+// If limit is less than 1, no limit is applied.
+// Note that the given records should already be ordered by the given columns.
 // WARNING: Only ascending ordering is currently supported.
 func MergeRecords(
-	mem memory.Allocator, records []arrow.Record, orderByCols []int,
+	mem memory.Allocator,
+	records []arrow.Record,
+	orderByCols []int,
+	limit int64,
```

Comment:
> If limit is less than 1, no limit is applied.

Can we just say `0` . I checked  `MergeRecords` call site the only cases with no limit uses `-1` . In this case `0` pretty much means no limit, we can leverage the type system and use `uint64`

```suggestion
	limit uint64,
```

Reply:
Sounds like a plan :+1: